### ISSUE TITLE
bcu not working with node 7.10

### ISSUE THREAD

**Issue by patrikx3** [tjunnone/npm-check-updates#345]
- [x ] I have read the list of [known issues](https://github.com/tjunnone/npm-check-updates#known-issues) before filing this issue.
- [ x] I have searched for [similiar issues](https://github.com/tjunnone/npm-check-updates/issues) before filing this issue.

* node version: 
current

* npm version: 
current

* npm-check-updates version: 
current

---------------------------
```(node:30637) DeprecationWarning: Calling an asynchronous function without callback is deprecated.```

**Comment by bourgois** [comment]
Same issue for me, just updated to 7.10 ...
@patrikx3 did you found a solution?

**Comment by karanjthakkar** [comment]
@bourgois Can you post what you see when you try `type ncu` and `type node` in the terminal?

**Comment by bourgois** [comment]
Sorry, I forget to come back here: it's solved. But I don't now what happened in fact, I opened a new terminal later and it worked. Probably some environment var needed to reset the terminal?

**Comment by oupala** [comment]
I've also upgraded to node 7.10.0 and I get the same error message.

I think it could be a good thing to reopen this issue.

**Comment by oupala** [comment]
```bash
$ type node
node est haché (/home/<user>/.nvm/versions/node/v7.10.0/bin/node)
$ type ncu
ncu est haché (/home/<user>/.nvm/versions/node/v7.10.0/bin/ncu)
```

Message is localized (in french).

**Comment by oupala** [comment]
Reverting node version to 7.9.0 makes ncu **works** again.

Upgrading node version to 8.0.0 makes ncu **don't work** again.

**Comment by raineorshine** [comment]
@oupala Can you post the full error message? I'm not seeing where it's failing.

**Comment by oupala** [comment]
@raineorshine I don't know what you mean by *the full error message*, because I posted the full error message:

```bash
$ ncu
Using /path/to/package.json
(node:21611) DeprecationWarning: Calling an asynchronous function without callback is deprecated.
```

I you want some more log, no problem, but tell me how to get the log you want.

**Comment by raineorshine** [comment]
@oupala Ah! I didn't realize that was the full error. Was hoping there'd be a stack trace. Now I know :). 

1. Can you describe what the behavior is after you see the error message? Does it exit immediately? Does it hang?
2. Can you run `ncu --loglevel verbose` and post here?

Thanks!

**Comment by oupala** [comment]
After writing the error, ncu does exit immediatly:

```bash
$ ncu
Using /home/romain/Documents/dev/recommendations/metalsmith/package.json
(node:12309) DeprecationWarning: Calling an asynchronous function without callback is deprecated.
$
```

Here is the output with the `verbose` option:

```bash
$ ncu --loglevel verbose
Initializing...
Running in local mode...
Finding package file data...
Using /home/romain/Documents/dev/recommendations/metalsmith/package.json
(node:12449) DeprecationWarning: Calling an asynchronous function without callback is deprecated.
$
```

There is not much information...

**Comment by raineorshine** [comment]
@oupala I have bad news. I tested this on node `v8` and it worked fine for me... so it appears to be platform/environment-specific. The worst kind of bug!

Without having a way to troubleshoot myself, you would have to insert console.logs throughout the source code in `/Users/raine/.nvm/versions/node/v8.0.0/lib/node_modules/npm-check-updates/lib/npm-check-updates.js` or wherever ncu is installed to find where it is crashing. I don't know if you're up for that!

I can tell you that it is crashing after the ["Using /path/to/package.json" message](https://github.com/tjunnone/npm-check-updates/blob/master/lib/npm-check-updates.js#L376) and before the ["Getting installed packages..." message](https://github.com/tjunnone/npm-check-updates/blob/master/lib/npm-check-updates.js#L178).

**Comment by oupala** [comment]
I'm up for that. Tell me where to add what debug message and I'll do that.

And, by the way, I love the way you started debugging:

> I can tell you that it is crashing after the "Using /path/to/package.json" message and before the "Getting installed packages..." message.

can be translated as:

> I can tell you that it is crashing after the line 376 and before the line 178.

Which looks counter-intuitive at first sight... :-)

**Comment by raineorshine** [comment]
@oupala Haha, yes!

So, I could tell you explicitly where to put debug messages, but then I would have to walk through every step with you. Maybe I can explain more generally the approach so it makes sense for you. I apologize ahead of time if this is all too obvious.

We have something that is crashing ncu without warning between the above-mentioned lines. We can find the problem location in logarithmic time by cutting the problem space in half each time, i.e. placing a console.log in the middle and seeing if it gets to it.

e.g.

1. WORKS
2. ?
3. ?
4. console.log('Am I there yet?')
5. ?
6. ?
7. NEVER GETS HERE

Then proceed to log at about the halfway point till you narrow in on what line is causing it to crash. Obviously this is complicated by the fact that there are different functions being called at different places so the bug could be spatially anywhere!

This might help you understand the general architecture!

![npm-check-updates architecture](https://files.gitter.im/npm-check-updates/Lobby/Gi7f/npm-check-updates-flow.png)

**Comment by reeslo** [comment]
I've same message : 
$ncu
(node:28480) DeprecationWarning: Calling an asynchronous function without callback is deprecated.

**Comment by theskillwithin** [comment]
same problem here

**Comment by oupala** [comment]
I've processed as you said, with node 8.0.0. It took me some time. But I wanted to report you where the problem was, it appeared that I wan not using the last version oc **ncu**.

```bash
$ ncu --version
2.12.1
```

I updated **ncu** to the last version:

```bash
$ ncu --version
2.12.1
```

And it does work well now with node v8.0.0.

So to all users that have this problem, try to update npm-check-updates and report here if it fixes the problem.

It did for me!
